### ISSUE TITLE
Update rsocket-bom packaging to pom

### ISSUE THREAD

**Issue by linux-china** [rsocket/rsocket-java#688]
Now rsocket-bom is jar type, not pom type on http://central.maven.org/maven2/io/rsocket/rsocket-bom/1.0.0-RC3/

The following dependency declaration under dependencyManagement does not work. 

```
 <dependency>
                <groupId>io.rsocket</groupId>
                <artifactId>rsocket-bom</artifactId>
                <version>1.0.0-RC3</version>
                <type>pom</type>
                <scope>import</scope>
</dependency>
```

**Comment by OlegDokuka** [comment]
@linux-china Yeah. I noticed that. Just fixed in RC4 snapshot. Should be better

**Comment by OlegDokuka** [comment]
https://oss.jfrog.org/oss-snapshot-local/io/rsocket/rsocket-bom/1.0.0-RC4-SNAPSHOT/rsocket-bom-1.0.0-RC4-20190902.064842-5.pom

**Comment by OlegDokuka** [comment]
fixed in #55b847b
